### PR TITLE
[MSCTFIME][SDK][USER32] Add msctfime.ime (stub)

### PR DESCRIPTION
**Pull request by katahiromz**
## Overview
1. `msctfime.ime` is an IME file interface for new-style IMEs a.k.a. "Text Input Processors" (TIPs).
2. `msctfime.ime` is loaded as old-style IME file at `ImmLoadLayout` in specific condition.
3. `msctfime.ime` communicates with the current TIP (This feature is not implemented yet). 

JIRA issue: [CORE-19360](https://jira.reactos.org/browse/CORE-19360)

## Proposed changes

- Add `msctfime` module at `dll/ime/msctfime`.
- The functions in this module are currently stub.
- Move IME file interface declarations from `<imm.h>` to `<ddk/immdev.h>`.
- Modify `ImmNotifyIME`, `NotifyIME`, and `ImeProcessKey` prototypes for x64 compliance.

## TODO

- [x] Do build.

### CODE REVIEW

File: dll/ime/msctfime/msctfime.rc [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#include <windef.h>
```
Comment (HBelusca) [inline dll/ime/msctfime/msctfime.rc:1]: Shouldn't we add the standard file headers also in the resource files (this one and the en-US.rc one)?
cc @binarymaster and others.

File: sdk/include/psdk/imm.h [hunk 2]
```diff
@@ -270,23 +270,6 @@ BOOL WINAPI
 ImmDestroySoftKeyboard(
     _In_ HWND hwndSoftKBD);
 
-BOOL WINAPI ImeInquire(LPIMEINFO, LPWSTR, LPCWSTR lpszOptions);
```
Comment (HBelusca) [inline sdk/include/psdk/imm.h:273]: You confirm that those "Ime*" functions are indeed not present in the official MS PSDK imm.h file?

Comment (katahiromz) [reply sdk/include/psdk/imm.h:273]: Yep, they should be in `immdev.h`.

File: dll/ime/msctfime/msctfime.c [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+ */
+
+#include "msctfime.h"
+
+WINE_DEFAULT_DEBUG_CHANNEL(msctfime);
+
+HINSTANCE g_hInst = NULL; /* The instance of this module */
+
+BOOL WINAPI
+ImeInquire(
```
Comment (HBelusca) [inline dll/ime/msctfime/msctfime.c:15]: Since this is brand new ReactOS-specific code, I would suggest you add SAL annotations to all the functions.

**Review by HBelusca — APPROVED** [review]
OK otherwise, please wait for @binarymaster to give feedback on the file headers thing.

File: dll/ime/msctfime/lang/en-US.rc [hunk 4]
```diff
@@ -0,0 +1,16 @@
+LANGUAGE LANG_ENGLISH, SUBLANG_ENGLISH_US
```
Comment (binarymaster) [inline dll/ime/msctfime/lang/en-US.rc:1]:
```suggestion
/*
 * PROJECT:     ReactOS msctfime.ime
 * LICENSE:     LGPL-2.1-or-later (https://spdx.org/licenses/LGPL-2.1-or-later)
 * PURPOSE:     English (United States) resource file
 * TRANSLATOR:  Copyright 2023 Katayama Hirofumi MZ <katayama.hirofumi.mz@gmail.com>
 */

LANGUAGE LANG_ENGLISH, SUBLANG_ENGLISH_US
```

File: dll/ime/msctfime/msctfime.c (hunk 3 above)
Comment (katahiromz) [reply dll/ime/msctfime/msctfime.c:15]: Added SAL annotations.

File: dll/ime/msctfime/msctfime.rc (hunk 1 above)
Comment (katahiromz) [reply dll/ime/msctfime/msctfime.rc:8]: Added.

**Pull request merged**